src/main/java/algorithmic_task/TaskService.java

```diff
@@ -0,0 +1,69 @@
+package algorithmic_task;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+public class TaskService {
+
+    private Task task;
+    private List<Set<String>> dictionary;
```

Comment:
What is the purpose of class Task? Are you sure that it is thread-safe?

Reply:
Let's make dictionary as a separate component which is initialized during start-up.

Reply:
I created Dictionary class, I think now it is thread safe.